src/components/map-marker/map-marker.tsx

```diff
@@ -81,20 +110,28 @@ export class MapMarker implements GxComponent {
       );
       this.markerInstance = marker([0, 0], {
         icon: divIcon({
-          iconAnchor: [halfIconWidth, this.iconHeight],
+          className: this.markerClass,
+          iconAnchor: [halfIconSizes.width, this.iconHeight],
+          popupAnchor: [0, -halfIconSizes.height],
           iconSize: [this.iconWidth, this.iconHeight],
-          tooltipAnchor: [0, -28]
+          tooltipAnchor: [0, -halfIconSizes.height]
         })
       });
     }
+    this.setPopup();
     if (this.tooltipCaption) {
-      this.markerInstance.bindTooltip(this.tooltipCaption);
+      this.markerInstance.bindTooltip(this.tooltipCaption, {
+        direction: "top"
+      });
     }
     this.gxMapMarkerDidLoad.emit(this.markerInstance);
   }
 
   componentDidUpdate() {
-    const halfIconWidth = this.iconWidth / 2;
+    const halfIconSizes = {
+      height: this.iconHeight / 2,
+      width: this.iconWidth / 2
+    };
```

Comment:
This is repeated in the `componentDidLoad` event. Can we encapsulate this?

Reply:
done in 7dfb8c6